### ISSUE TITLE
Allow creation of gestures  set as a modifers for add-ons

### ISSUE THREAD

**Issue by lukaszgo1** [nvaccess/nvda#11233]
Apologies for the vague title - it is difficult to describe my intended use case in a single sentence.

### Is your feature request related to a problem? Please describe.
Some add-ons have a lot of gestures assigned to some modifiers and numbers.  Examples include:
- SwitchSynth for switching between slots.
- Columns review for reading column at the given index.
- Win Wizard for hiding window at the current slot.

In all these cases only part of the gesture that differs is the digit pressed with the modifiers.
There are various approaches in these add-ons to allow these gestures to be customized by the user:
- Win Wizard - no customization at all.
- Switch Synth - each of these 6 scripts has to be reassigned individually if user wishes to change the base modifiers.
- Columns Review - the base modifiers can be changed from the  settings dialog.

These solutions are either not very user friendly or are not integrated well with the Input gestures dialog.

### Describe the solution you'd like
I propose addition of a new parameter to the script decorator which would specify that the given gesture consists of base modifiers.
In practice the gesture would need to be send as is to the OS when pressed alone or with keys which are not specified by the add-on author and execute the desired function otherwise.
### Describe alternatives you've considered
I haven't considered any so feedback is appreciated.
### Additional context
It might be useful for built-in appmodules for Miranda and Vipmud.

**Comment by leonardder** [comment]
To clarify, I think table navigation is a good example here. You navigate tables with ctrl+alt+arrow keys. If you want to reassign these gestures, you have to reassign the four of them, whereas in your proposal, it should be possible to change the modifiers to something else which changes all the four of them instantly. Is this correct?

**Comment by lukaszgo1** [comment]
Yes, it is. I haven't thought about table navigation for some reason.

**Comment by XLTechie** [comment]
I am having a hard time imagining how this would work in practice.

Can you give a hard example of some example commands, and how you would remap 
them if gestures dialog implemented this feature in some way?

**Comment by XLTechie** [comment]
Also, please correct the spelling of "modifers" to "modifiers" in the title.

Maybe a more English version of this title would be:

Allow creation of gesture sets with a modifier for add-ons

Or, perhaps encapsulating better what I think you want to do here:

Allow creation of a group of gestures with a single configurable set of modifiers

**Comment by CyrilleB79** [comment]
I agree that changing a whole set of gestures may a bit tedious just for a modifier. Anyway it is an action to perform only once.
I fear that UX in the gesture definition may be degraded. Let's see anyway if a more precise proposal is made regarding the gesture dialog behaviour. For example, identifying conflicting gestures for the user may be more difficult than today.

**Comment by CyrilleB79** [comment]
An other example add-on is Outlook Extended. The strategy is to define individual gestures: alt+1, ..., alt+0, alt+-, alt+= and let the user change them one by one if required.
